### PR TITLE
feat: add auth

### PR DESCRIPTION
**Pull request by vovchanchyk**
- add basic auth rout

### CODE REVIEW

File: src/controllers/auth.controller.ts [hunk 1]
```diff
@@ -0,0 +1,12 @@
+import { Request, Response } from 'express';
+import { authService } from '../services/auth.service';
+
+export const register = async (req: Request, res: Response) => {
+  const tokens = await authService.register(req.body);
```
Comment (R1KO) [inline src/controllers/auth.controller.ts:5]: where is a validation?

File: src/services/auth.service.ts [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+    if (!user) {
+      throw new Error('user not found');
+    }
+    const match = await bcrypt.compare(password, user.password);
+    if (!match) {
+      throw new Error('wrong password');
+    }
+
+    const accessToken = jwt.sign(
```
Comment (R1KO) [inline src/services/auth.service.ts]: I think generate tokens should be another service

File: src/services/auth.service.ts [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+      { name, email },
```
Comment (R1KO) [inline src/services/auth.service.ts]: it's bad info for store i token, better pass only id. And you can pass email only id you really need it

File: src/services/auth.service.ts [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+      process.env.ACCESS_TOKEN_SECRET as Secret,
+      { expiresIn: '10m' },
```
Comment (R1KO) [inline src/services/auth.service.ts]: 10m-1h it's ok period

File: src/services/auth.service.ts [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+    );
+
+    const refreshToken = jwt.sign(
+      { name, email },
+      process.env.REFRESH_TOKEN_SECRET as Secret,
+      { expiresIn: '2h' },
```
Comment (R1KO) [inline src/services/auth.service.ts]: better use longer period as 1 week or 1 month. When this token is expired, a user must be login again

File: src/services/auth.service.ts [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+  async register(userData: ICreateUser) {
+    const { password, name, email } = userData;
+    const hashedPassword = await bcrypt.hash(password, 10);
+    await this.userService.createUser({
+      name,
+      email,
+      password: hashedPassword,
+    });
+
+    const accessToken = jwt.sign(
```
Comment (R1KO) [inline src/services/auth.service.ts]: it's a duplicate code. move to separated method

File: src/services/auth.service.ts [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+      { name, email },
+      process.env.ACCESS_TOKEN_SECRET as Secret,
+      { expiresIn: '10m' },
```
Comment (R1KO) [inline src/services/auth.service.ts]: move literals to consts & env

File: src/types/IUser.ts [hunk 8]
```diff
@@ -2,9 +2,13 @@ export interface IUser {
   id: number;
   name: string;
   email: string;
+  password: string;
 }
 
 export interface ICreateUser {
```
Comment (R1KO) [inline src/types/IUser.ts:8]: ICreateUserDto

File: src/controllers/auth.controller.ts (hunk 1 above)
Comment (vovchanchyk) [reply src/controllers/auth.controller.ts:5]: addressed

**Review by R1KO — APPROVED** [review]

File: src/services/auth.service.ts (hunk 2 above)
Comment (vovchanchyk) [reply src/services/auth.service.ts]: addressed

File: src/services/auth.service.ts (hunk 3 above)
Comment (vovchanchyk) [reply src/services/auth.service.ts]: addressed

File: src/types/IUser.ts (hunk 8 above)
Comment (vovchanchyk) [reply src/types/IUser.ts:8]: i will skip it for while, i need time for it

File: src/services/jwt.service.ts [hunk 9]
```diff
@@ -0,0 +1,33 @@
+import jwt, { Secret } from 'jsonwebtoken';
+
+export class JwtService {
+  private createToken(id: string, secret: string, expiresIn: string) {
```
Comment (R1KO) [inline src/services/jwt.service.ts:4]: I think better to make this method more universal.
The best way for that - pass object like {key: value}.

type Payload = {
[key:string]: any;
};

**Pull request merged**